salt/pwyf-dqt.sls

```diff
@@ -0,0 +1,168 @@
+## Publish What You Fund Data Quality Tester
+## https://github.com/pwyf/data-quality-tester
+
+{% from 'lib.sls' import createuser, apache, uwsgi %}
+
+##### System install
+
+include:
+  - core
+  - apache
+  - uwsgi
+#   - letsencrypt
+
+# uWSGI from data-quality-tester > requirements.txt has a C
+# extension not sure if this is needed given we install the pkg
+# uwsgi-plugin-python3 but it is in the current requirements so we
+# need to make sure it is installable
+
+pwyf-dqt-deps:
+  pkg.installed:
+    - pkgs:
+      - virtualenv
+      - libapache2-mod-proxy-uwsgi
+      - uwsgi-plugin-python3
+      - redis-server
+      # See uWSGI comment ^
+      - build-essential
+      - python3-dev
+    - watch_in:
+      - service: apache2
+      - service: uwsgi
+
+  apache_module.enabled:
+    - name: proxy proxy_uwsgi
+    - watch_in:
+      - service: apache2
+      - service: uwsgi
+
+# redis should start up after being installed but this will verify
+
+redis-server:
+  pkg:
+    - installed
+  service:
+    - running
+    - enable: True
+    - reload: True
+
+{{ createuser(pillar.pwyf_dqt.user) }}
+
+##### Git checkout 
+
+/home/{{ pillar.pwyf_dqt.user }}/{{ pillar.pwyf_dqt.checkout_dir }}:
+  git.latest:
+    - name: {{ pillar.pwyf_dqt.git_url }}
+    - rev: {{ pillar.pwyf_dqt.branch }}
+    - user: {{ pillar.pwyf_dqt.user }}
+    - target: /home/{{ pillar.pwyf_dqt.user }}/{{ pillar.pwyf_dqt.checkout_dir }}
+    - force_fetch: True
+    - force_reset: True
+    - submodules: True
+    - require:
+      - pkg: git
+
+##### Install python deps in virtualenv
+
+/home/{{ pillar.pwyf_dqt.user }}/{{ pillar.pwyf_dqt.checkout_dir }}/.ve:
+  virtualenv.managed:
+    - python: /usr/bin/python3
+    - user: {{ pillar.pwyf_dqt.user }}
+    - system_site_packages: False
+    - requirements: /home/{{ pillar.pwyf_dqt.user }}/{{ pillar.pwyf_dqt.checkout_dir }}/requirements.txt
+    - require:
+      - git: /home/{{ pillar.pwyf_dqt.user }}/{{ pillar.pwyf_dqt.checkout_dir }}
+
+##### Flask app setup
+
+setup_dqt:
+  cmd.run:
+    - runas: {{ pillar.pwyf_dqt.user }}
+    - env:
+      - FLASK_APP: 'DataQualityTester/__init__.py'
+      - FLASK_ENV: 'development'
```

Comment:
Is this temporary or is this staying at "dev"? Either way a comment to explain might be good.

Reply:
Thanks yeah, I'll remove that